master/buildbot/test/unit/test_steps_source_git.py

```diff
@@ -1225,7 +1225,7 @@ def test_mode_full_clean_no_existing_repo_branch(self):
     def test_mode_full_clean_no_existing_repo_with_origin(self):
         self.setupStep(
             self.stepClass(repourl='http://github.com/buildbot/buildbot.git',
-                           mode='full', method='clean', origin='foo'))
+                           mode='full', method='clean', origin='foo', progress=True))
```

Comment:
Why do we add progress param to most tests? This will ensure that we don't test the code path when progress is not set.

Reply:
We should instead leave the setup part of most tests as is, update the changed tests with new behavior and also add a test for checking whether explicit progress=True is disabled on old git and whether progress=False disables `--progress` on new it.